pkg/controller/integration/kits.go

```diff
@@ -257,10 +259,32 @@ func matchesTrait(it map[string]interface{}, kt map[string]interface{}) bool {
 }
 
 func hasMatchingSources(it *v1.Integration, kit *v1.IntegrationKit) bool {
-	if len(it.Sources()) != len(kit.Spec.Sources) {
+	// A kit may have fewer sources during native builds but should not have more
+	if len(kit.Spec.Sources) > len(it.Sources()) {
```

Comment:
We may remove this check completely. It would make sense in case of a clear not equality only IMO.